src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncBlockHeadersValidator.scala

```diff
@@ -0,0 +1,58 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.event.LoggingAdapter
+import io.iohk.ethereum.consensus.validators.Validators
+import io.iohk.ethereum.domain.{ BlockHeader, Blockchain }
+import io.iohk.ethereum.network.Peer
+
+trait FastSyncBlockHeadersValidator {
+  import FastSyncBlockHeadersValidator._
+
+  def blockchain: Blockchain
+  def validators: Validators
+  def log: LoggingAdapter
+
+  /** Validates received header
+    *
+    * @param header                   header to validate
+    * @param peer                     peer which send the header
+    * @param nextBlockToFullyValidate block number of the next block to validate fully
+    *
+    * @return the valid blockHeader with flag which tells if validationState should be updated
+    *         or the error encountered during validation
+    */
+  def validateHeader(
+    header: BlockHeader,
+    peer: Peer,
+    nextBlockToFullyValidate: BigInt
+  ): Either[HeaderProcessingResult, (BlockHeader, Boolean)] = {
+    if (header.number >= nextBlockToFullyValidate) {
+      validators.blockHeaderValidator.validate(header, blockchain.getBlockHeaderByHash) match {
+        case Right(_) =>
+          Right((header, true))
+
+        case Left(error) =>
+          log.warning("Block header validation failed during fast sync at block {}: {}", header.number, error)
+          Left(ValidationFailed(header, peer))
+      }
+    } else {
+      Right((header, false))
+    }
+  }
+
+  def areHeadersFormingChain(headers: Seq[BlockHeader]): Boolean = {
```

Comment:
I think it would require to be extracted somewhere else and make Ledger use extracted version; I did extract it too in my PR with Regular Sync.

Reply:
maybe extracting this to some other place right now is not a good idea. I would prefer to wait for your PR to be merged and then do this change here, wdyt?